CONTRIBUTING.md

```diff
@@ -1,21 +1,129 @@
-# Contributing
+# How to contribute
 
-## Evolving the Format of the Forecast Data
+Soaringmeteo team is glad you are reading this, because we always need volunteers
+to keep this awesome service up to date.
 
-The frontend consumes the forecast data produced by the backend, so both parties need to agree on their format.
-Furthermore, the frontend also reads several forecast data in the past (users can show the data from the latest
-forecast run, or from older runs as well). This has consequences on how the format of the forecast data produced
-by the backend can evolve:
+* Find a bug? Do not hesitate to open an issue using the bug template. **TODO: LINK HERE**
+* Have an awesome improvement idea? Do not hesitate to open an issue using the feature template **TODO: LINK HERE**
 
-- We can add or remove optional fields,
-- We can remove non-optional fields (the frontend will simply ignore those fields on the older forecast data).
+# Setup the development environment
 
-However, to add non-optional fields we have to perform the following procedure:
+The following steps assumes that you are running a debian based OS, and has been tested under Ubuntu 20.04 LTS.
+
+## Tools install for the  backend
+
+1. java
+
+```bash
+sudo apt install openjdk-17-jre
+sudo apt install openjdk-17-jdk
+# to confirm
+java --version
+```
+
+2. sbt
+
+```bash
+echo "deb https://repo.scala-sbt.org/scalasbt/debian all main" | sudo tee /etc/apt/sources.list.d/sbt.list
+echo "deb https://repo.scala-sbt.org/scalasbt/debian /" | sudo tee /etc/apt/sources.list.d/sbt_old.list
+curl -sL "https://keyserver.ubuntu.com/pks/lookup?op=get&search=0x2EE0EA64E40A89B84B2DF73499E82A75642AC823" | sudo apt-key add
+sudo apt update
+sudo apt install sbt
+```
+
+More informations are available [here](https://scala-sbt.org).
+
+## Tools install for the  frontend
+
+1. yarn3 (instructions taken from [here](https://yarnpkg.com/getting-started/install)
+```bash
+corepack enable
+yarn set version 3
+yarn --version
+```
+
+2. node18 (managed through  nvm)
+```
+curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.5/install.sh | bash
+
+nvm install 18
+```
+
+## Build and run the backend
+
+The backend `gfs` and `wrf` must be build and run to prepare the assets consumed by the frontend.
+
+1. Move to backend directory 
+```bash
+cd backend
+```
+2. Start sbt server 
+```
+sbt
+```
+3. Build the `gfs` program: 
+
+```
+gfs/Universal/packageZipTarball
+tar -xzf soaringmeteo-gfs.tgz
+```
+
+4. Build the `wrf` program:
+
+```
+wrf/Universal/packageZipTarball
+tar -xzf soaringmeteo-wrf.tgz
+```
```

Comment:
Actually, this step is not necessary, the tasks `makeGfsAssets` and `makeWrfAssets` automatically compile the program before running it.